packages/firestore/src/core/firestore_client.ts

```diff
@@ -522,6 +522,20 @@ export class FirestoreClient {
     });
   }
 
+  /**
+   * Returns a `Promise` resolves when all the pending writes at the time when this method is called
```

Comment:
I would remove the backticks for Promise. It's just a normal term in JS, not a class or method name (also it should be "Promise that resolves");

Also: s/when all the pending writes at the time/when all writes that were pending at the time/